mobile/src/user/UserRepository.ts

```diff
@@ -0,0 +1,48 @@
+import type { User } from "./UserTypes";
+
+export interface UserRepository {
+  signIn(name: string): Promise<User>;
+  signOut(): Promise<void>;
+  me(): Promise<User | null>;
+}
+
+export class RemoteUserRepository {
```

Comment:
This is a static class (it has no `this`), so you don't really need a class for that. An object literal will do. 

```js
const remoteObjectRepository = {
   signIn: async () => /** ... */,
   signOut: async () => /** ... */,
   me: async () => /** ... */,
};
```

Notice that I changed the name to use a lowercase initial.